mesonpy/__init__.py

```diff
@@ -704,6 +708,17 @@ def _validate_metadata(self) -> None:
                     f'expected `{self._metadata.requires_python}`'
                 )
 
+    def _check_for_unknown_config_keys(self, valid_args: Mapping[str, Collection[str]]) -> None:
```

Comment:
I would have written this as static method (or a function) accepting the config settings (the result of `self._config.get('tool', {}).get('meson-python', {})`) and the description of the valid configuration keys.

Reply:
Something like
```python
@staticmethod
def _check_config(config: dict, schema: Type):
    valid_args = typing_get_args(schema)
    ...
```
```

Reply:
That requires us to change the implementation when we add new keys, so I am not really a fan, besides we are just passing around data that the class would already have access to.

Reply:
This is essentially what we have, but without it being a static method. The class would already have access to the config anyways, and we are not parsing different configs, so I don't see the point in doing it like this :sweat_smile:

Reply:
I agree. In my opinion either this is made generic to check a configuration dict against a scheme, or it is made to do not take any arguments at all. Now it is an in-between that IMHO does not make much sense.

Reply:
So, essentially, you want this to also check the types, instead of just the key structure?

Reply:
That is not the main point I was trying to make, but if it makes sense, why not?

Reply:
I think it makes sense, but might be worth doing in a future PR. I really wanted to merge this and then finally tag a release :blush:

Reply:
Sure. Can we have the documentation updates back up and running before the release? Otherwise the published documentation will be outdated. We also need documentation for the `pyproject.toml` settings.

Reply:
The docs are working, they just aren't reporting back to Github for some reason I haven't managed to figure out :confused: 

https://readthedocs.org/projects/meson-python/builds/

Reply:
This can happen after moving repos. Often removing and reinstalling the GitHub hooks/integration helps.